pkg/scheduler/framework/plugins/names/names.go

```diff
@@ -30,10 +30,6 @@ const (
 	NodeResourcesFit                = "NodeResourcesFit"
 	NodeUnschedulable               = "NodeUnschedulable"
 	NodeVolumeLimits                = "NodeVolumeLimits"
-	AzureDiskLimits                 = "AzureDiskLimits"
-	CinderLimits                    = "CinderLimits"
-	EBSLimits                       = "EBSLimits"
-	GCEPDLimits                     = "GCEPDLimits"
```

Comment:
This change could be a breaking change for users who explicitly uses those plugins in the scheduler config.

@kubernetes/sig-scheduling-approvers How do we usually deal with plugin removal? Just removing while mentioning in the release note? Or do we have to follow a deprecation process or something?

Reply:
Those plugins have been deprecated by https://github.com/kubernetes/kubernetes/pull/74544 in 1.14.

Reply:
BTW, I update this PR's description.

Fixes https://github.com/kubernetes/kubernetes/issues/82375

Reply:
Got it, can you add more information on the release note? e.g., do users of those plugins have to migrate to something else? Or no action is required at all?

Reply:
I think there's no required action caused by this PR. Because those volume types are always translated to the CSI type. If the node status contains volume limitation info, the csi scheduler plugin will use it even though the csinode doesn't include this info.

FYI: https://github.com/kubernetes/kubernetes/blob/6673e7a93d8f0e81343309350e3ec7e7afc7fd73/pkg/scheduler/framework/plugins/nodevolumelimits/csi.go#L406-L409

Reply:
Oh, I forgot that I also changed the CSI plugin in this PR. It wouldn't read limits from the node object anymore. If a storage vendor wants to use the limits, they have to create manually the CSI driver object when deploying the driver. It should be a standard way to deploy a CSI driver if limits are needed. So I want to know whether I need to mention this in the release note. If so, I will add it in the release note. cc @sanposhiho what do you think?

Reply:
> I think there's no other required action caused by this PR. Because those volume types are always translated to the CSI type. If the node status contains volume limitation info, the csi scheduler plugin will use it even though the csinode doesn't include this info.

OK, then please mention it; e.g., `Because XXXX, no action is required for users for this removal. Please just remove those plugins if you explicitly use those plugins in the scheduler config.`

>  I want to know whether I need to mention this in the release note. If so, I will add it in the release note. cc @sanposhiho what do you think?

I believe that's worth enough to be mentioned in the release note, but defer to someone from sig-storage as I don't have much context around that.

Reply:
cc @jsafrane

Reply:
I agree with @carlory, Azure, GCE, AWS and Cinder Limits, using data from Node objects are not used any longer. These volume plugins were migrated to CSI and CSINode object is used instead. These plugins should not do anything in v1.30.

What does kube-scheduler process do with unknown plugins specified in its config file? If it exits, then 
this PR could break clusters who explicitly enabled/disabled the plugins there. 

> If a storage vendor wants to use the limits, they have to create manually the CSI driver object when deploying the driver. It should be a standard way to deploy a CSI driver if limits are needed.

It's actually automatic. A CSI driver reports how much volumes can a node handle in [NodeGetInfoResponse](https://github.com/container-storage-interface/spec/blob/2696773158b970c65107610d8f3ef4302546eca0/csi.proto#L1689) and kubelet stores this limit in CSINode. Scheduler then knows the driver's limit on the node. And each node can have a different limit.

Reply:
> What does kube-scheduler process do with unknown plugins specified in its config file? If it exits, then
this PR could break clusters who explicitly enabled/disabled the plugins there.

It exits -
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/runtime/framework.go#L628

But, according to https://github.com/kubernetes/kubernetes/pull/74544 @carlory shew, those plugins are already marked as deprecated. So, it should be OK to just remove them as long as we mention it in the release note, I believe.

Reply:
@sanposhiho the release note is updated.

Reply:
@jsafrane Please take a look at the release note from sig-storage pov too.

Reply:
Can we remove these plugins from default plugins first and make them invalid the same time: https://github.com/kubernetes/kubernetes/blob/9c9a0e3ba57cb103d651e77f1b9283673c4d0cd7/pkg/scheduler/apis/config/validation/validation.go#L101-L106 This is how we used to deprecate plugins, then next release let's remove the total logics together, I know these plugins are already marked as deprecated for releases, but let's be less surprised to users as remove plugins from default ones directly.

If so let's mention that **Those plugins are removed from default plugins as they're already marked as deprecated since 1.14 and will be totally removed in v1.31, also mention the replacement we have**

Or we're blocked by these plugins like schedulingHint?

Reply:
/hold 
merge it in 1.32

Reply:
/un-lgtm 
for now

Reply:
/remove-lgtm